src/index.ts

```diff
@@ -152,7 +153,15 @@ export class Client {
         },
         this.httpHeaders,
       )
-      .then(response => response.methodResponses[0][1]);
+      .then(response => {
+        const methodResponse = response.methodResponses[0];
+
+        if (methodResponse[0] === 'error') {
+          throw methodResponse[1];
```

Comment:
What is being thrown here exactly? It is unclear from code.

Reply:
According to the type definition, it should be a ResponseType. But here it will throw an error body. I have some work in progress about types, but I did not want to include it in this pull request, and I am not actively working on this project currently.

Reply:
I see. Will this PR stay open until you will become more active on given project again?

Reply:
I want this PR to be merged, even with this type problem, because the error case was not handled before.

Reply:
@alagane would you like me or @andrevka to make it more clear or just approve this one and make it more clear with separate PR? @chibenwa any thoughts regarding this?

Reply:
This is likely open for too long. I agree merging this, and of course welcome improvements.